fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -476,6 +476,13 @@ public void commitTransaction(long transactionId, List<TabletCommitInfo> tabletC
             if (table == null) {
                 throw new MetaNotFoundException("Table does not exist: " + tableId);
             }
+            boolean hasRunningJobs =
+                    GlobalStateMgr.getCurrentState().getInsertOverwriteJobManager().hasRunningOverwriteJob(
+                            transactionId, tableId, tableToPartition.get(tableId).stream().collect(Collectors.toList()));
+            if (hasRunningJobs) {
```

Comment:
should we check whether hasRunningJobs in here?

Reply:
I find only here I can get the partitions to commit. So I check it here

Reply:
I think this logic is a little strange. Why until one transaction is going to commit, we just check there running insert jobs.  I think we could check it early.